### PR TITLE
Remove captain from mining director

### PR DESCRIPTION
**Pull request by NullSoldier**
This removes captain usage from mining director, and moves the mining
event it had on it, into the director itself.

### CODE REVIEW

File: ironfish/src/mining/director.test.slow.ts [hunk 1]
```diff
@@ -51,7 +52,7 @@ describe('Mining director', () => {
   let targetSpy: jest.SpyInstance
   let targetMeetsSpy: jest.SpyInstance
   let isAddBlockValidSpy: jest.SpyInstance
-  let memPool: TestMemPool
+  let mempool: TestMemPool
```
Comment (dguenther) [inline ironfish/src/mining/director.test.slow.ts:55]: I think if we want to do casing of the `mempool` variable this way (either is fine by me), the class should be cased `TestMempool` and `Mempool`

File: ironfish/src/mining/director.test.slow.ts [hunk 2]
```diff
@@ -325,17 +320,13 @@ describe('Recalculating target', () => {
     jest.setTimeout(15000000)
     captain = await makeCaptain(strategy)
     memPool = new MemPool(captain)
```
Comment (dguenther) [inline ironfish/src/mining/director.test.slow.ts:322]: here's another differing memPool casing

**Review by dguenther — APPROVED** [review]

File: ironfish/src/mining/director.test.slow.ts (hunk 1 above)
Comment (NullSoldier) [reply ironfish/src/mining/director.test.slow.ts]: reverted

File: ironfish/src/mining/director.test.slow.ts (hunk 2 above)
Comment (NullSoldier) [reply ironfish/src/mining/director.test.slow.ts:324]: reverted

**Pull request merged**